src/test/java/com/cdancy/bitbucket/rest/features/BranchApiMockTest.java

```diff
@@ -100,6 +114,33 @@ public void testListBranches() throws Exception {
         }
     }
 
+
+    private static class BuildStatusInfo {
```

Comment:
The below works as expected. No need to do any converting as jclouds behind the scenes will just pass along the `JsonElement` should you request it. I like this approach better as it allows for variable returned data and the user can then decided how they want to handle the json.

```
    public abstract Map<String, JsonElement> metadata();

    Branch() {
    }

    @SerializedNames({ "id", "displayId", "type", "latestCommit", "latestChangeset", "isDefault", "metadata", "errors" })
    public static Branch create(String id, String displayId, String type,
                                String latestCommit, String latestChangeset, 
                                boolean isDefault, Map<String, JsonElement> metadata, List<Error> errors) {
        
        return new AutoValue_Branch(Utils.nullToEmpty(errors), id, displayId, type, 
                latestCommit, latestChangeset, isDefault, Utils.nullToEmpty(metadata));
    }
```
You'll have to make a change to the mockTest you wrote but other than that this will work.